src/wire-protocol.ts

```diff
@@ -16,7 +16,7 @@ import {
 /**
  * A request to make to Fauna.
  */
-export interface QueryRequest extends QueryRequestHeaders {
```

Comment:
@cleve-fauna We have already changed the signature of Client.query to accept a `Query` (previously `QueryBuilder`) and not `Query | QueryRequest`. That makes `QueryRequest` specifically just the thing we send over the wire, and it doesn't make sense for it to also have the headers fields. (We were probably sending the options over the wire for a time, weren't we 🤔 )

It also mean that the note for `Client.query` that "you can embed header fields in this object; if you do that there's no need to pass the headers parameter." is no longer accurate... I didn't catch that before now.

As a side-effect, type enforcement for `QueryRequest` and `QueryRequestOptions` in methods like `Client.query` and `Client.#setHeaders` is more accurate.